server/planning/feeding_services/onclusive_api_service.py

```diff
@@ -33,48 +39,58 @@ def _update(self, provider, update):
         :type update: dict
         :return: a list of events which can be saved.
         """
-        URL = app.config["ONCLUSIVE_URL"]
-        USERNAME = app.config["ONCLUSIVE_USERNAME"]
-        PASSWORD = app.config["ONCLUSIVE_PASSWORD"]
-        TOKEN = app.config["ONCLUSIVE_TOKEN"]
-        REF_TOKEN = app.config["ONCLUSIVE_REF_TOKEN"]
-
+        session = requests.Session()
+        current_date = datetime.now()
+        URL = provider["config"]["url"]
+        USERNAME = provider["config"]["username"]
+        PASSWORD = provider["config"]["password"]
+        TOKEN = ""
+        REF_TOKEN = ""
         # authntication get token
         if not TOKEN:
-            auth_url = URL + "api/v2/auth"
-            body = {"username": USERNAME, "password": PASSWORD}
-            resp = self.session.post(auth_url, body)
-
-            resp.raise_for_status()
-            if resp.status_code == 200 and resp.content:
-                data = resp.json()
-                if data.get("token") and data.get("refreshToken"):
-                    TOKEN = data["token"]
-                    REF_TOKEN = data["refreshToken"]
+            try:
+                auth_url = URL + "api/v2/auth"
+                body = {"username": USERNAME, "password": PASSWORD}
+                resp = session.post(auth_url, body)
+
+                resp.raise_for_status()
+                if resp.status_code == 200 and resp.content:
+                    data = resp.json()
+                    if data.get("token") and data.get("refreshToken"):
+                        TOKEN = data["token"]
+                        REF_TOKEN = data["refreshToken"]
+            except Exception as e:
+                logger.error("Not Found URL {}".format(URL))
```

Comment:
not sure if the error is correct, could be different issues if you catch Exception

Reply:
What if we handle it something like credentials auth errors or we can just log the exception:
```
except Exception:
            raise CredentialsAuthError(credentials)
```

Reply:
I think for catching HTTP exceptions the best way is:
```
except requests.HTTPError as http_error:
    logger.error("Authntication Failed {}".format(http_error))
except Exception as e:
    logger.error("Exception: {}".format(e))
```
This will provide the exact error message/exception message.

Reply:
yep but I actually wonder why catching it, you want the update to stop on such exception not continue right?

Reply:
Yes, we actually want to stop the update, but I see what you mean here, you mean to say that we should just raise the exception which will just stop the execution?

Reply:
yep any exception will pretty much stop it. so unless we want to handle it in some specific way we can just raise that or rather not catch it